pkg/xds/secrets/secrets.go

```diff
@@ -22,22 +24,33 @@ import (
 
 var log = core.Log.WithName("xds").WithName("secrets")
 
+type MeshCa struct {
+	Mesh     string
+	CaSecret *core_xds.CaSecret
+}
+
 type Secrets interface {
-	GetForDataPlane(dataplane *core_mesh.DataplaneResource, mesh *core_mesh.MeshResource) (*core_xds.IdentitySecret, *core_xds.CaSecret, error)
-	GetForZoneEgress(zoneEgress *core_mesh.ZoneEgressResource, mesh *core_mesh.MeshResource) (*core_xds.IdentitySecret, *core_xds.CaSecret, error)
+	GetForDataPlane(dataplane *core_mesh.DataplaneResource, mesh *core_mesh.MeshResource, otherMeshes []*core_mesh.MeshResource) (*core_xds.IdentitySecret, []MeshCa, error)
+	GetForGatewayListener(mesh *core_mesh.MeshResource, dataplane *core_mesh.DataplaneResource, listenerTags map[string]string, otherMeshes []*core_mesh.MeshResource) (*core_xds.IdentitySecret, MeshCa, error)
```

Comment:
You're right, at the moment there will be different identity certs for different listeners, which as is will lead to a bug since they have the same name? But in fact maybe this is unnecessary and all listeners on the same DP can share an identity cert

Reply:
now no longer dependent on gateway/listener tags